test/kitchen/test/integration/win-agent-with-cws-option/rspec_datadog/win-agent-with-cws-option_spec.rb

```diff
@@ -0,0 +1,131 @@
+    it 'has system probe service installed' do
+        expect(is_windows_service_installed("datadog-system-probe")).to be_truthy
+    end
+
+    # We retrieve the value defined in kitchen.yml because there is no simple way
+    # to set env variables on the target machine or via parameters in Kitchen/Busser
+    # See https://github.com/test-kitchen/test-kitchen/issues/662 for reference
+
+    # the `let` has to be inside an it block.
+    let(:expect_cws_installed) {
+        parse_dna().fetch('dd-agent-rspec').fetch('cws_included')
+    }
+
+    it 'has properly installed driver files' do
+        if expect_cws_installed
+            expect(is_windows_service_installed("datadog-security-agent")).to be_truthy
+            expect(is_windows_service_installed("ddprocmon")).to be_truthy
+
+            program_files = safe_program_files
+            expect(File).to exist("#{program_files}\\DataDog\\Datadog Agent\\bin\\agent\\ddprocmon.cat")
+            expect(File).to exist("#{program_files}\\DataDog\\Datadog Agent\\bin\\agent\\ddprocmon.sys")
+            expect(File).to exist("#{program_files}\\DataDog\\Datadog Agent\\bin\\agent\\ddprocmon.inf")
+        else
+            expect(is_windows_service_installed("datadog-security-agent")).to be_falsey
+            expect(is_windows_service_installed("ddprocmon")).to be_falsey
+
+            program_files = safe_program_files
+            expect(File).not_to exist("#{program_files}\\DataDog\\Datadog Agent\\bin\\agent\\ddprocmon.cat")
+            expect(File).not_to exist("#{program_files}\\DataDog\\Datadog Agent\\bin\\agent\\ddprocmon.sys")
+            expect(File).not_to exist("#{program_files}\\DataDog\\Datadog Agent\\bin\\agent\\ddprocmon.inf")
+
+        end
+    end
+    it 'does not have the driver running on install' do
+        ## verify that the driver is not started yet
+        expect(is_service_running?("ddprocmon")).to be_falsey
+    end
+
+end
+  
+shared_examples_for 'a Windows Agent with CWS running' do
+    # We retrieve the value defined in kitchen.yml because there is no simple way
+    # to set env variables on the target machine or via parameters in Kitchen/Busser
+    # See https://github.com/test-kitchen/test-kitchen/issues/662 for reference
+    let(:expect_cws_installed) {
+        parse_dna().fetch('dd-agent-rspec').fetch('cws_included')
+    }
+    it 'can start security agent' do
+        if expect_cws_installed
+            sa_conf_path = ""
+            sp_conf_path = ""
+            if os != :windows
+                sa_conf_path = "/etc/datadog-agent/security-agent.yaml"
+                sp_conf_path = "/etc/datadog-agent/system-probe.yaml"
+            else
+                sa_conf_path = "#{ENV['ProgramData']}\\Datadog\\security-agent.yaml"
+                sp_conf_path = "#{ENV['ProgramData']}\\Datadog\\system-probe.yaml"
+            end
+            f = File.read(sa_conf_path)
+            confYaml = YAML.load(f)
+            if !confYaml
```

Comment:
Why do we proceed with the remainder of the test if there isn't a valid `confYaml`? If we expect CWS to be installed, then the `.yaml` files should have been written already and if they aren't, then shouldn't the test fail here? Otherwise it seem we're artificially allowing the test to continue after we realize the files aren't there.

Big caveat in that I'm not completely well versed with kitchen - on my first read though, it seems like this should be a fail case.

Reply:
I suppose we could make it a fail case.  But we don't have "warnings" or "fail and continue".
So since we can continue in this case and see that the service starts...
But, i see your point and happy to change.

Reply:
Thanks, yeah, I think that we should make this (and any other checks that are required before we'd start the service) a fail case. We could separate it out into different functions like:

```
if cws_expected:
    if !verifyCWSFilesWritten(): // check they exist at the expected paths
        fail("expected files not found)

    if !verifyCWSFileContent(): // check they have the right key/values
       fail("malformed default files")

    enableCWSInFiles() // change whatever flags we need to enabled: true

    // continue to check that the services started up  
```

Reply:
I put in the checks for existence.
However, I left the existing error checking code in place.  Our default files for both of these are entirely commented out, which means that they're effectively an empty file.  So even if the file is present as expected, we have to build the correct config entrires.

Reply:
Ah that's annoying - are there tests somewhere else validate the default config is correct before everything is commented out? I mainly want to make sure that the default (albeit commented out) config is correct so that when customers comment out the key/values, the commented out key/values are correct.